### PR TITLE
OSDOCS-6703: Documented the minimum required vCenter privileges by OC…

### PR DESCRIPTION
**Pull request by dfitzmau**
[OSDOCS-6703](https://issues.redhat.com/browse/OSDOCS-6703)

Version(s):
4.15

CHeck out the vmware vSphere CSI Driver Operator requirements and vCenter requirements sections in the following documents:

Link to docs preview:
* [vCenter requirements IPI](https://68470--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/ipi/ipi-vsphere-installation-reqs#installation-vsphere-installer-infra-requirements_ipi-vsphere-installation-reqs)
* [vCenter requirements UPI](https://68470--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/upi/upi-vsphere-installation-reqs#installation-vsphere-installer-infra-requirements_upi-vsphere-installation-reqs)
* [VMware vSphere CSI Driver Operator requirements - IPI](https://68470--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/ipi/ipi-vsphere-installation-reqs#vsphere-csi-driver-reqs_ipi-vsphere-installation-reqs)
* [VMware vSphere CSI Driver Operator requirements - UPI](https://68470--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/upi/upi-vsphere-installation-reqs#vsphere-csi-driver-reqs_upi-vsphere-installation-reqs)

SME review:
- [x] SME has approved this change.

QE review:
- [x] QE has approved this change.

Additional information:
* [SPLAT-1103](https://issues.redhat.com/browse/SPLAT-1103)
* [OSDOCS-7538](https://issues.redhat.com/browse/OSDOCS-7538)

### CODE REVIEW

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 1]
```diff
@@ -376,6 +376,343 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
...
+`VirtualMachine.Interact.Reset`
+`VirtualMachine.Inventory.Create`
+`VirtualMachine.Inventory.CreateFromExisting`
+`VirtualMachine.Inventory.Delete`
+`VirtualMachine.Provisioning.Clone`
+`VirtualMachine.Provisioning.MarkAsTemplate`
+`VirtualMachine.Provisioning.DeployTemplate`
+
+|vSphere vCenter Datacenter
+|If the installation program creates the virtual machine folder. For UPI, VirtualMachine.Inventory.Create and VirtualMachine.Inventory.Delete privileges are optional if your cluster does not use the Machine API.
```
Comment (dfitzmau) [inline modules/installation-vsphere-installer-infra-requirements.adoc:502]: UPI has no relevance here?

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 2]
```diff
@@ -376,6 +376,343 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
...
+`VirtualMachine.Config.Settings`
+`VirtualMachine.Interact.PowerOff`
+`VirtualMachine.Interact.PowerOn`
+`VirtualMachine.Inventory.CreateFromExisting`
+`VirtualMachine.Inventory.Delete`
+`VirtualMachine.Provisioning.Clone`
+`VirtualMachine.Provisioning.DeployTemplate`
+
+|vSphere vCenter Datacenter
+|If the installation program creates the virtual machine folder. For UPI, VirtualMachine.Inventory.Create and VirtualMachine.Inventory.Delete privileges are optional if your cluster does not use the Machine API.
```
Comment (dfitzmau) [inline modules/installation-vsphere-installer-infra-requirements.adoc:619]: UPI has no relevance here?

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 3]
```diff
@@ -376,6 +376,343 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
...
```
Comment (dfitzmau) [inline modules/installation-vsphere-installer-infra-requirements.adoc:705]: This section is irrelivant as it goes against the intent of the API machine table?

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 4]
```diff
@@ -376,6 +376,343 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
...
+`VirtualMachine.Inventory.Delete`
+`VirtualMachine.Provisioning.Clone`
+`VirtualMachine.Provisioning.DeployTemplate`
+`VirtualMachine.Provisioning.MarkAsTemplate`
+`Folder.Create`
+`Folder.Delete`
+|===
+====
+
+.Minimum permissions for post-installation management of components
```
Comment (dfitzmau) [inline modules/installation-vsphere-installer-infra-requirements.adoc:540]: Post installation here applies to IPI and UPI?

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 5]
```diff
@@ -376,6 +376,343 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
 
 For more information about creating an account with only the required privileges, see link:https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.security.doc/GUID-5372F580-5C23-4E9C-8A4E-EF1B4DD9033E.html[vSphere Permissions and User Management Tasks] in the vSphere documentation.
 
+
+[discrete]
+[id="installation-vsphere-installer-infra--minimum-requirements_{context}"]
+== Minimum required vCenter account privileges
+
+You can create a custom role with privilege sets that propogate a minimum set of permissions to each vSphere object. Depending on the installation stage, you might need to restrict access to certain vSphere objects.
```
Comment (dfitzmau) [inline modules/installation-vsphere-installer-infra-requirements.adoc:384]: A user can create a custom role through the vCenter Client or is some sort of CLI option available?

https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-security/GUID-18071E9A-EED1-4968-8D51-E0B4F526FDA3.html

Comment (dfitzmau) [reply modules/installation-vsphere-installer-infra-requirements.adoc:384]: I see a note in this link that states "When you add a custom role and do not assign any privileges to it, the role is created as a Read Only", so I should state this, because the cloud provider minimum privileges is based around this default"?

File: modules/installation-vsphere-installer-infra-requirements.adoc (hunk 4 above)
Comment (rvanderp3) [reply modules/installation-vsphere-installer-infra-requirements.adoc:540]: yes

File: modules/installation-vsphere-installer-infra-requirements.adoc (hunk 3 above)
Comment (rvanderp3) [reply modules/installation-vsphere-installer-infra-requirements.adoc:705]: i think keeping this is fine, but maybe link to the machine API privileges section

File: modules/installation-vsphere-installer-infra-requirements.adoc (hunk 2 above)
Comment (rvanderp3) [reply modules/installation-vsphere-installer-infra-requirements.adoc:619]: i think keeping this is fine, but maybe link to the machine API privileges section

Comment (dfitzmau) [reply modules/installation-vsphere-installer-infra-requirements.adoc:619]: Link to the machine API for the purposes of the IPI?

**Comment by dfitzmau** [comment]
Hi @mjpytlak . Putting a comment here for a reminder in the new year as a consideration for the updates proposed on this PR. The minimum privlege tables impact both IPI and UPI vSphere routes, so should the **vCenter requirements** section be moved to the **Preparing to install on vSphere** book? This book seems like a shared location for content that touches both UPI and IPI.

**Comment by mjpytlak** [comment]
Thanks for the heads up @dfitzmau. Per the Miro board [1]  I am using to refactor the vSphere content, the module that you are updating (`modules/vmware-csi-driver-reqs.adoc`) will be used in two net new installation requirements assemblies. One for IPI and one for UPI. Check out Proposed vSphere docs Information Architecture. The new assemblies will appear under Installer-provisioned infrastructure and User-provisioned infrastructre.

[1] https://miro.com/app/board/uXjVNbUhuKQ=/

File: modules/installation-vsphere-installer-infra-requirements.adoc (hunk 3 above)
Comment (dfitzmau) [reply modules/installation-vsphere-installer-infra-requirements.adoc]: Updated for numerous tables.

File: modules/installation-vsphere-installer-infra-requirements.adoc (hunk 5 above)
Comment (dfitzmau) [reply modules/installation-vsphere-installer-infra-requirements.adoc]: Thanls, @rvanderp3 . Updated. I'll move the PR to the next review cycle.

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 6]
```diff
@@ -129,7 +129,7 @@ endif::upi[]
 `VirtualMachine.Provisioning.DeployTemplate`
 
 |vSphere vCenter Datacenter
-|If the installation program creates the virtual machine folder. For UPI, `VirtualMachine.Inventory.Create` and `VirtualMachine.Inventory.Delete` privileges are optional if your cluster does not use the Machine API.
+|If the installation program creates the virtual machine folder. For user-provisioned infrastructure, `VirtualMachine.Inventory.Create` and `VirtualMachine.Inventory.Delete` privileges are optional if your cluster does not use the Machine API. See the Minimum permissions for the Machine API table.
```
Comment (kcarmichael08) [inline modules/installation-vsphere-installer-infra-requirements.adoc:132]:
```suggestion
|If the installation program creates the virtual machine folder. For user-provisioned infrastructure, `VirtualMachine.Inventory.Create` and `VirtualMachine.Inventory.Delete` privileges are optional if your cluster does not use the Machine API. See the "Minimum permissions for the Machine API" table. 
```
Elsewhere in table, quotes are used in this reference for this table so I would be consistent.

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 7]
```diff
@@ -3,11 +3,11 @@
 // * installing/installing_vsphere/ipi/ipi-vsphere-installation-reqs.adoc
 // * installing/installing_vsphere/upi/upi-vsphere-installation-reqs.adoc
```
Comment (kcarmichael08) [inline modules/installation-vsphere-installer-infra-requirements.adoc:5]: I think it is missing the mod docs content type field, ID, and module title: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#module-file-metadata

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 8]
```diff
@@ -361,6 +362,348 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
 
 For more information about creating an account with only the required privileges, see link:https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.security.doc/GUID-5372F580-5C23-4E9C-8A4E-EF1B4DD9033E.html[vSphere Permissions and User Management Tasks] in the vSphere documentation.
 
+
+[discrete]
+[id="installation-vsphere-installer-infra--minimum-requirements_{context}"]
+== Minimum required vCenter account privileges
+
+After you create a custom role and assign privileges to it, you can create permissions by selecting specific vSphere objects and then assign the custom role to a user or group for each object. Depending on the installation stage, you might need to restrict access to certain vSphere objects.
```
Comment (kcarmichael08) [inline modules/installation-vsphere-installer-infra-requirements.adoc:370]:
```suggestion
After you create a custom role and assign privileges to it, you can create permissions by selecting specific vSphere objects and then assigning the custom role to a user or group for each object. Depending on the installation stage, you might need to restrict access to certain vSphere objects. 
```

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 9]
```diff
@@ -361,6 +362,348 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
...
+[IMPORTANT]
+====
+If you create a custom role and you do not assign privileges to it, the vSphere Server by default assigns a `Read Only` role to the custom role.
+====
+
+Consider creating a custom role when an account with global administrative privileges does not meet your needs. By using a custom role, you can choose to meet the minimum set of privileges necessary for {product-title} cluster installation. 
+
+[IMPORTANT]
+====
+Installing an {product-title} cluster in a vCenter is tested against a full list of privileges as described in the section named _"Required vCenter account privileges"_. By adhering to the full list of privileges, you can reduce the possibility of unexpected behaviors that might occur when creating a custom role with a restricted set of privileges. Red Hat does not support an environment that is configured to meet a minimum set of permissions.
```
Comment (kcarmichael08) [inline modules/installation-vsphere-installer-infra-requirements.adoc:381]:
```suggestion
Installing an {product-title} cluster in a vCenter is tested against a full list of privileges as described in the "Required vCenter account privileges" section. By adhering to the full list of privileges, you can reduce the possibility of unexpected behaviors that might occur when creating a custom role with a restricted set of privileges. Red Hat does not support an environment that is configured to meet a minimum set of permissions. 
```

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 10]
```diff
@@ -361,6 +362,348 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
...
+====
+
+Consider creating a custom role when an account with global administrative privileges does not meet your needs. By using a custom role, you can choose to meet the minimum set of privileges necessary for {product-title} cluster installation. 
+
+[IMPORTANT]
+====
+Installing an {product-title} cluster in a vCenter is tested against a full list of privileges as described in the section named _"Required vCenter account privileges"_. By adhering to the full list of privileges, you can reduce the possibility of unexpected behaviors that might occur when creating a custom role with a restricted set of privileges. Red Hat does not support an environment that is configured to meet a minimum set of permissions. 
+====
+
+Propogating a minimum set of permissions to specific vSphere objects, such as `vSphere vCenter Resource Pool`, is situational based. For more details about these situations, see the proceeding tables.
```
Comment (kcarmichael08) [inline modules/installation-vsphere-installer-infra-requirements.adoc:384]:
```suggestion
Propagating a minimum set of permissions to specific vSphere objects, such as `vSphere vCenter Resource Pool`, is situationally based. For more details about these situations, see the preceding tables.
```

File: modules/vmware-csi-driver-reqs.adoc [hunk 11]
```diff
@@ -21,3 +21,133 @@ If a third-party vSphere CSI driver is present in the cluster, {product-title} d
 ====
 The VMware vSphere CSI Driver Operator is supported only on clusters deployed with `platform: vsphere` in the installation manifest.
 ====
+
+You can create a custom role for the Container Storage Interface (CSI) and the CSI Driver Operator. The custom role can include privilege sets that assign a minimum set of permissions to each vSphere object. This means that the Container Storage Interface (CSI) and CSI Driver Operator can establish a basic interaction with these objects. The following tables detail these minimum set of permissions.
+
+[IMPORTANT]
+====
+Installing an {product-title} cluster in a vCenter is tested against a full list of privileges as described in the section named _"Required vCenter account privileges"_. By adhering to the full list of privileges, you can reduce the possibility of unexpected and unsupported behaviors that might occur when creating a custom role with a set of restricted privileges.
```
Comment (kcarmichael08) [inline modules/vmware-csi-driver-reqs.adoc:29]:
```suggestion
Installing an {product-title} cluster in a vCenter is tested against a full list of privileges as described in the "Required vCenter account privileges" section. By adhering to the full list of privileges, you can reduce the possibility of unexpected and unsupported behaviors that might occur when creating a custom role with a set of restricted privileges. 
```

**Review by kcarmichael08 — COMMENTED** [review]
Some minor suggestions/comments - otherwise lgtm! Nicely done!

File: modules/installation-vsphere-installer-infra-requirements.adoc (hunk 7 above)
Comment (kcarmichael08) [reply modules/installation-vsphere-installer-infra-requirements.adoc:5]: Sorry, it's not missing the ID - I see it below, but the title needs to be a Level 1 heading.

**Comment by sgaoshang** [comment]
lgtm for the installer part @duanwei33 Could you help to take a look at CSI part? Thanks.
[VMware vSphere CSI Driver Operator requirements - IPI](https://68470--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/ipi/ipi-vsphere-installation-reqs#vsphere-csi-driver-reqs_ipi-vsphere-installation-reqs)
[VMware vSphere CSI Driver Operator requirements - UPI](https://68470--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/upi/upi-vsphere-installation-reqs#vsphere-csi-driver-reqs_upi-vsphere-installation-reqs)

**Comment by duanwei33** [comment]
@sgaoshang @dfitzmau Sure, I will check, but may I know where do these privileges come from? May I have the link?

**Comment by sgaoshang** [comment]
> @sgaoshang @dfitzmau Sure, I will check, but may I know where do these privileges come from? May I have the link?

@duanwei33 Pls refer to [doc](https://gist.github.com/rvanderp3/4158de02a6ba3dd72eb454a4fdbd8706) in [SPLAT-1228](https://issues.redhat.com/browse/SPLAT-1228)

File: modules/vmware-csi-driver-reqs.adoc [hunk 12]
```diff
@@ -21,3 +21,133 @@ If a third-party vSphere CSI driver is present in the cluster, {product-title} d
...
+|vSphere object for role
+|When required
+|Required privileges
+
+|vSphere vCenter
+|Always
+|
+[%hardbreaks]
+`Cns.Searchable`
+`Read Only`
```
Comment (duanwei33) [inline modules/vmware-csi-driver-reqs.adoc:46]: Hi @dfitzmau, I might miss something, but if we need some privileges, why do we need "Read Only" here? I think "Read Only" is only applicable for no specific privilege needed like "vSphere Port Group" in Line 74.

Comment (dfitzmau) [reply modules/vmware-csi-driver-reqs.adoc:46]: Thanks, @duanwei33 . @rvanderp3 , if a vSphere object requires minimum permissions that extends beyond `Read Only`, do we need to list `Read Only`?

File: modules/vmware-csi-driver-reqs.adoc [hunk 13]
```diff
@@ -21,3 +21,133 @@ If a third-party vSphere CSI driver is present in the cluster, {product-title} d
...
 ====
+
+You can create a custom role for the Container Storage Interface (CSI) and the CSI Driver Operator. The custom role can include privilege sets that assign a minimum set of permissions to each vSphere object. This means that the Container Storage Interface (CSI) and CSI Driver Operator can establish a basic interaction with these objects. The following tables detail these minimum set of permissions.
+
+[IMPORTANT]
+====
+Installing an {product-title} cluster in a vCenter is tested against a full list of privileges as described in the "Required vCenter account privileges" section. By adhering to the full list of privileges, you can reduce the possibility of unexpected and unsupported behaviors that might occur when creating a custom role with a set of restricted privileges. 
+====
+
+.Minimum permissions for the CSI
```
Comment (duanwei33) [inline modules/vmware-csi-driver-reqs.adoc:32]: @dfitzmau @rvanderp3 Can we use a "Storage" which contains `CSI Driver` +  `CSI Driver Operator` + `vsphere problem detector` (we missed here)? Does it make more sense? Because we can disable "Storage" capability but not for each of them, and it would make things simple.

File: modules/vmware-csi-driver-reqs.adoc [hunk 14]
```diff
@@ -21,3 +21,79 @@ If a third-party vSphere CSI driver is present in the cluster, {product-title} d
...
 ====
+
+You can create a custom role for the Container Storage Interface (CSI) driver, the vSphere CSI Driver Operator, and the vSphere Problem Detector Operator. The custom role can include privilege sets that assign a minimum set of permissions to each vSphere object. This means that the CSI driver, the vSphere CSI Driver Operator, and the vSphere Problem Detector Operator can establish a basic interaction with these objects. The following table details these minimum set of permissions.
+
+[IMPORTANT]
+====
+Installing an {product-title} cluster in a vCenter is tested against a full list of privileges as described in the "Required vCenter account privileges" section. By adhering to the full list of privileges, you can reduce the possibility of unexpected and unsupported behaviors that might occur when creating a custom role with a set of restricted privileges. 
+====
+
+.Minimum permissions for the storage components
```
Comment (duanwei33) [inline modules/vmware-csi-driver-reqs.adoc:32]: I think this should be added at the same level with "Minimum permissions for the Machine API", under the section "Minimum required vCenter account privileges".

Comment (dfitzmau) [reply modules/vmware-csi-driver-reqs.adoc]: Thanks, @duanwei33 . Updated. I moved the storage table among the other tables.

File: modules/vmware-csi-driver-reqs.adoc (hunk 13 above)
Comment (dfitzmau) [reply modules/vmware-csi-driver-reqs.adoc]: Hi @duanwei33 . I merged these tables into 1 table. WDYT?

File: modules/vmware-csi-driver-reqs.adoc (hunk 12 above)
Comment (dfitzmau) [reply modules/vmware-csi-driver-reqs.adoc]: Hi @duanwei33 . Updated. Does the content LGTY?

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 15]
```diff
@@ -361,6 +363,383 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
...
+After you create a custom role and assign privileges to it, you can create permissions by selecting specific vSphere objects and then assigning the custom role to a user or group for each object. 
+
+Before you create permissions or request for the creation of permissions for a vSphere object, determine what minimum permissions apply to the vSphere object. By doing this task, you can ensure a basic interaction exists between a vSphere object and {product-title} architecture. 
+
+[IMPORTANT]
+====
+If you create a custom role and you do not assign privileges to it, the vSphere Server by default assigns a `Read Only` role to the custom role. Note that for the cloud provider API, the custom role only needs to inherit the privileges of the `Read Only` role.
+====
+
+Consider creating a custom role when an account with global administrative privileges does not meet your needs. Also consider that depending on the installation stage, you might need to restrict access to certain vSphere objects.
```
Comment (rvanderp3) [inline modules/installation-vsphere-installer-infra-requirements.adoc:380]:
```suggestion
Consider creating a custom role when an account with global administrative privileges does not meet your needs.
```

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 16]
```diff
@@ -361,6 +363,383 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
...
+
+[IMPORTANT]
+====
+Installing an {product-title} cluster in a vCenter is tested against a full list of privileges as described in the "Required vCenter account privileges" section. By adhering to the full list of privileges, you can reduce the possibility of unexpected behaviors that might occur when creating a custom role with a restricted set of privileges. Red Hat does not support an environment that is configured to meet a minimum set of permissions.
```
Comment (rvanderp3) [inline modules/installation-vsphere-installer-infra-requirements.adoc:384]: Installing an {product-title} cluster in a vCenter is tested against a full list of privileges as described in the "Required vCenter account privileges" section. By adhering to the full list of privileges, you can reduce the possibility of unexpected behaviors that might occur when creating a custom role with a restricted set of privileges. Red Hat does not support an environment that is configured to meet a minimum set of permissions. 

maybe reword `Red Hat does not support an environment that is configured to meet a minimum set of permissions. ` to `Only an account configured with the required privileges is supported. `

Comment (dfitzmau) [reply modules/installation-vsphere-installer-infra-requirements.adoc]: Thanks. Updated.

Comment (duanwei33) [reply modules/installation-vsphere-installer-infra-requirements.adoc]: That sounds reasonable.

**Comment by duanwei33** [comment]
LGTM from storage part.

File: installing/installing_vsphere/ipi/ipi-vsphere-installation-reqs.adoc [hunk 17]
```diff
@@ -17,6 +17,7 @@ include::modules/vmware-csi-driver-reqs.adoc[leveloffset=+1]
 
 * To remove a third-party vSphere CSI driver, see xref:../../../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-install-issues_persistent-storage-csi-vsphere[Removing a third-party vSphere CSI Driver].
 * To update the hardware version for your vSphere nodes, see xref:../../../updating/updating_a_cluster/updating-hardware-on-nodes-running-on-vsphere.adoc#updating-hardware-on-nodes-running-on-vsphere[Updating hardware on nodes running in vSphere].
+* To list of minimum permissions for CSI and its components, see the "Minimum permissions for the storage components" table in the xref:../../installing_vsphere/ipi/ipi-vsphere-installation-reqs#installation-vsphere-installer-infra-requirements_ipi-vsphere-installation-reqs[Required vCenter account privileges] section.
```
Comment (ousleyp) [inline installing/installing_vsphere/ipi/ipi-vsphere-installation-reqs.adoc:20]: For this entire additional resources section (not necessarily in the scope of this PR), please see the [guidelines for additional resources sections](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#additional-resources-sections) in the OCP docs contribution guide. In particular:

> - The links and xrefs in the unordered list must contain minimal-length, human-readable text between the square brackets. This text is often the title of the linked page.
> - You must not include text outside of the square brackets.

This additional resource has a typo in it ("To list of"), so it's worth fixing even if you don't adjust the rest of the section. Because we're talking about a table, it's tricky, but maybe something like this?

```suggestion
* xref:../../installing_vsphere/ipi/ipi-vsphere-installation-reqs#installation-vsphere-installer-infra-requirements_ipi-vsphere-installation-reqs[The "Minimum permissions for the storage components" table in the "Minimum required vCenter account privileges" section]
```

File: installing/installing_vsphere/upi/upi-vsphere-installation-reqs.adoc [hunk 18]
```diff
@@ -16,6 +16,7 @@ include::modules/vmware-csi-driver-reqs.adoc[leveloffset=+1]
 
 * To remove a third-party vSphere CSI driver, see xref:../../../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-install-issues_persistent-storage-csi-vsphere[Removing a third-party vSphere CSI Driver].
 * To update the hardware version for your vSphere nodes, see xref:../../../updating/updating_a_cluster/updating-hardware-on-nodes-running-on-vsphere.adoc#updating-hardware-on-nodes-running-on-vsphere[Updating hardware on nodes running in vSphere].
+* To list of minimum permissions for CSI and its components, see the "Minimum permissions for the storage components" table in the xref:../../installing_vsphere/upi/upi-vsphere-installation-reqs#installation-vsphere-installer-infra-requirements_upi-vsphere-installation-reqs[Required vCenter account privileges] section.
```
Comment (ousleyp) [inline installing/installing_vsphere/upi/upi-vsphere-installation-reqs.adoc:19]: Please refer to my comment in `installing/installing_vsphere/ipi/ipi-vsphere-installation-reqs.adoc`, which also applies here :)

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 19]
```diff
@@ -361,6 +363,383 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
 
 For more information about creating an account with only the required privileges, see link:https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.security.doc/GUID-5372F580-5C23-4E9C-8A4E-EF1B4DD9033E.html[vSphere Permissions and User Management Tasks] in the vSphere documentation.
 
+
+[discrete]
+[id="installation-vsphere-installer-infra--minimum-requirements_{context}"]
```
Comment (ousleyp) [inline modules/installation-vsphere-installer-infra-requirements.adoc:368]: Is this ID correct? The `--` in the middle seems unintentional.

File: modules/installation-vsphere-installer-infra-requirements.adoc [hunk 20]
```diff
@@ -361,6 +363,383 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
...
+[IMPORTANT]
+====
+If you create a custom role and you do not assign privileges to it, the vSphere Server by default assigns a `Read Only` role to the custom role. Note that for the cloud provider API, the custom role only needs to inherit the privileges of the `Read Only` role.
+====
+
+Consider creating a custom role when an account with global administrative privileges does not meet your needs.
+
+[IMPORTANT]
+====
+Red Hat supports only an account that is configured with the required privileges. Installing an {product-title} cluster in a vCenter is tested against a full list of privileges as described in the "Required vCenter account privileges" section. By adhering to the full list of privileges, you can reduce the possibility of unexpected behaviors that might occur when creating a custom role with a restricted set of privileges.
```
Comment (ousleyp) [inline modules/installation-vsphere-installer-infra-requirements.adoc:384]: This phrasing is sitting oddly with me. I think changing "supports only" to "only supports" would help, but here's another suggestion to take or leave:
```suggestion
Accounts that are not configured with the required privileges are unsupported. Installing an {product-title} cluster in a vCenter is tested against a full list of privileges as described in the "Required vCenter account privileges" section. By adhering to the full list of privileges, you can reduce the possibility of unexpected behaviors that might occur when creating a custom role with a restricted set of privileges. 
```

**Review by ousleyp — COMMENTED** [review]
I think this is looking really good, though I didn't have time to carefully check all of the previews during this review. I just had a few comments. Please put the merge-review-needed label back on the PR when you are ready. Thanks so much, Darragh!

File: modules/installation-vsphere-installer-infra-requirements.adoc (hunk 19 above)
Comment (dfitzmau) [reply modules/installation-vsphere-installer-infra-requirements.adoc]: Fixed. I spotted this when changing the xref.

File: modules/installation-vsphere-installer-infra-requirements.adoc (hunk 20 above)
Comment (dfitzmau) [reply modules/installation-vsphere-installer-infra-requirements.adoc:382]: Good suggestion. I was trying to remove much passive voice as possible but I was wrestling with the word "only"!

File: installing/installing_vsphere/upi/upi-vsphere-installation-reqs.adoc (hunk 18 above)
Comment (dfitzmau) [reply installing/installing_vsphere/upi/upi-vsphere-installation-reqs.adoc]: OK. I added IDs and now the xref points directly to the table. All works well in the preview.

File: installing/installing_vsphere/ipi/ipi-vsphere-installation-reqs.adoc (hunk 17 above)
Comment (dfitzmau) [reply installing/installing_vsphere/ipi/ipi-vsphere-installation-reqs.adoc]: OK. I added IDs and now the xref points directly to the table. All works well in the preview.

File: modules/installation-vsphere-installer-infra-requirements.adoc (hunk 20 above)
Comment (ousleyp) [reply modules/installation-vsphere-installer-infra-requirements.adoc]: I think this (<thing> is unsupported) is one of the times when passive voice makes more sense, somehow. But I appreciate your effort to avoid it :)

File: installing/installing_vsphere/ipi/ipi-vsphere-installation-reqs.adoc (hunk 17 above)
Comment (ousleyp) [reply installing/installing_vsphere/ipi/ipi-vsphere-installation-reqs.adoc]: Love it!

**Comment by ousleyp** [comment]
LGTM! Really nice work. Merging now.
